File: 2015/10.py

import math, copy

with open('10_input') as f:
	data = f.read()
	# data = data.split()
	# data = list(map(int, data.split()))

def part_1():
	string = data
	x = 0
	while x < 50:
		# print(string)
		new_string = ''
		curr_len = 1
		for i in range(len(string) - 1):
			if string[i] == string[i + 1]:
				curr_len += 1
				continue 
			else:
				new_string += str(curr_len) + string[i]
				if i == len(string) - 2:
					new_string += '1' + str(string[-1])
				curr_len = 1

		string = new_string
		new_string = ''
		x += 1
	print(len(string))
	return

def part_2():
	return 


if __name__ == '__main__':
	part_1()
	#part_2()
	

File: 2015/10.js

const fs = require('fs');
const { exec } = require("child_process");
const { bin, float, hex, int, num, oct } = require("../lib");
const { range, drange, trange, iter, diter, titer } = require("../lib");
const { copy,	entries, in_, inBB, keys, sort, values } = require("../lib");
const { Counter, DD, empty, FastQueue, PriorityQueue, Queue, set, Stack } = require("../lib");
const { ord, chr, count, debug, disp, crt, gcd, lcm, modPow, mod, prod, prodBig, randint, sum, sumBig, transpose } = require("../lib");
const { digits, ints,	floats,	singles,	words,	lines,	table,	groups,	getGroups,	groupsWith,	parse,	parseLine } = require ('../lib');
const { min, max, random, abs, ceil, floor, log, log10, log2, round, sign, sin, cos, tan, asin, acos, atan, atan2, sqrt, PI } = Math;
const { isSuperset, or, and, xor, sub } = set;
const { getExecStr } = require("../lib/post");
const { next_permutation } = require("../lib/itertools");
// const { parsePacket, ops, HEX, ALU, VARS, resetVars } = require('./lib_2021'); // specific to AOC 2021

function part1(data) {

	range(50)(_ => {
		let tmp = '';
		let i = 0;
		while (i < data.length) {
			let j = i;
			let curr = data[i];
			while (i < data.length && curr === data[++i]);
			tmp += `${i-j}${curr}`;
		}
		data = tmp;
		if (_ === 39) {
			debug(data.length);
			console.log("END OF PART1");
		}
	})

	let res = data.length;
	debug(res);
	exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART2");
	return;
}

function part2(data) {

	let res;

	debug(res);
	// exec(`echo ${res} | xclip -sel clip -rmlastnl`);
	console.log("END OF PART2");
	return;
}

function main() {
	let data = fs.readFileSync("10_input").toString("utf-8");

	part1(data);
	// part2(data);
	process.exit(0);
}

main();
